### DISCUSSION TITLE
General Questions and Q & A About the Project

### DISCUSSION THREAD

**Discussion by nam20485** [nam20485/OdbDesign#152] [Q&A]
Feel free to ask here if you have any questions!

**Comment by ComputerCats** [comment]
Good aftertoon! Could I convert ODB++ in STL with your program?

**Comment by nam20485** [comment]
Sorry, this codebase does not contain any STL functionality.

**Comment by snhobbs** [comment]
So I got the server running and I'm seeing the swagger page which is all working and all looks promising! If possible a quick example usage snippet would be super useful. Looking forward to seeing what this can do, thanks!

**Reply by nam20485** [reply]
> So I got the server running and I'm seeing the swagger page which is all working and all looks promising! If possible a quick example usage snippet would be super useful. Looking forward to seeing what this can do, thanks!

You are quite welcome! Always happy to see people using the code.

A more full featured frontend is currently in the works.

For now to see some simple usage examples accessing the data  model directly from the library interface in C++ you can look at the [console app's main() function](https://github.com/nam20485/OdbDesign/blob/04cfbbceefdaaa10e581806e14e8d2f7213be0f6/OdbDesignApp/OdbDesignApp.cpp#L79-L144). 

Depending on which data you are after (or all of it) different endpoints from the REST API can be used and the data will be provided in the response in JSON data. 

Great idea. I 'll add a simple walkthru for how to load and parse a design in an ODB++ archive and then fetch the parsed data back in JSON. Should be posted soon.
